src/Controls/tests/Xaml.UnitTests/Issues/GhXSGDeadCode.xaml.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Linq;
+using Microsoft.Maui.Controls.Core.UnitTests;
+using Microsoft.Maui.Graphics;
+using NUnit.Framework;
+
+using static Microsoft.Maui.Controls.Xaml.UnitTests.MockSourceGenerator;
+
+namespace Microsoft.Maui.Controls.Xaml.UnitTests;
+
+public partial class GhXSGDeadCode : ContentPage
+{
+	public GhXSGDeadCode() => InitializeComponent();
+
+	[TestFixture]
+	class Tests
+	{
+		[SetUp] public void SetUp() => Application.Current = new MockApplication();
+
+		[Test]
+		public void SetterPropertyUsesCompiledConverter([Values] XamlInflator inflator)
+		{
+			var page = new GhXSGDeadCode(inflator);
+			var style = (Style)page.Resources["testStyle"];
+			
+			Assert.NotNull(style);
+			Assert.AreEqual(2, style.Setters.Count);
+			
+			var fontSizeSetter = style.Setters[0];
+			Assert.AreEqual(Label.FontSizeProperty, fontSizeSetter.Property);
+			Assert.AreEqual(16.0, fontSizeSetter.Value);
+			
+			var colorSetter = style.Setters[1];
+			Assert.AreEqual(Label.TextColorProperty, colorSetter.Property);
+			Assert.AreEqual(Colors.Red, colorSetter.Value);
+		}
+
+		[Test]
+		public void SetterPropertyDoesNotGenerateDeadCode([Values(XamlInflator.SourceGen)] XamlInflator inflator)
```

Comment:
This test doesn't need to run for each inflator. The parameter is not used an it will just run 3 times doing the same. Thinking about this some more, I belive this test would fit better into the `SourceGen.UnitTests.csproj` project.